nox/_parametrize.py

```diff
@@ -13,9 +13,59 @@
 # limitations under the License.
 
 import functools
+import itertools
 
 
-def parametrize_decorator(arg_names, arg_values_list):
+class Param:
+    """A class that encapsulates a single set of parameters to a parametrized
+    session."""
+
+    def __init__(self, *args, arg_names=None, id=None):
+        self.args = tuple(args)
+        self.id = id
+
+        if arg_names is None:
+            arg_names = ()
+
+        self.arg_names = tuple(arg_names)
+
+    @property
+    def call_spec(self):
+        return dict(zip(self.arg_names, self.args))
+
+    def __str__(self):
+        if self.id:
+            return self.id
+        else:
+            call_spec = self.call_spec
+            keys = sorted(call_spec.keys(), key=str)
+            args = ["{}={}".format(k, repr(call_spec[k])) for k in keys]
+            return ", ".join(args)
+
+    __repr__ = __str__
+
+    def copy(self):
+        new = self.__class__(*self.args, arg_names=self.arg_names, id=self.id)
+        return new
+
+    def update(self, other):
+        self.args = self.args + other.args
+        self.arg_names = self.arg_names + other.arg_names
+        self.id = ", ".join([str(self), str(other)])
+
+    def __eq__(self, other):
+        if isinstance(other, self.__class__):
+            return (
+                self.args == other.args
+                and self.arg_names == other.arg_names
+                and self.id == other.id
+            )
+        elif isinstance(other, dict):
+            return dict(zip(self.arg_names, self.args)) == other
+        return False
```

Comment:
Shouldn't this be `return NotImplemented`?

Reply:
Good catch. It's been a while since I've done this.